remote-api/src/main/java/org/alfresco/repo/webdav/WebDAVLockServiceImpl.java

```diff
@@ -237,68 +236,87 @@ public Void execute() throws Throwable
         }
     }
 
-    public void lock(NodeRef nodeRef, LockInfo lockInfo)
+    public void lock(NodeRef nodeRef, LockInfo lockInfo) {
+        int timeout = (int) lockInfo.getRemainingTimeoutSeconds();
+        lock(nodeRef, lockInfo, timeout);
+    }
+
+    /**
+     * Shared method for webdav/vti protocols to lock node. If node is locked for more than 24 hours it is automatically added
+     * to the current session locked resources list.
+     *
+     * @param nodeRef the node to lock
+     * @param userName userName
+     * @param timeout the number of seconds before the locks expires
+     */
+    @Override
+    public void lock(NodeRef nodeRef, String userName, int timeout)
+    {
+        LockInfo lockInfo = createLock(nodeRef, userName, true);
+        lock(nodeRef, lockInfo, timeout);
+    }
+
+    public void lock(NodeRef nodeRef, LockInfo lockInfo, int timeout)
     {
-        boolean performSessionBehavior = false;
-        long timeout;
-        
-        timeout = lockInfo.getRemainingTimeoutSeconds();    
-        
         // ALF-11777 fix, do not lock node for more than 24 hours (webdav and vti)
-        if (timeout >= WebDAV.TIMEOUT_24_HOURS || timeout == WebDAV.TIMEOUT_INFINITY)
-        {   
+        boolean performSessionBehavior = false;
+        if (timeout > WebDAV.TIMEOUT_24_HOURS || timeout == WebDAV.TIMEOUT_INFINITY)
+        {
             timeout = WebDAV.TIMEOUT_24_HOURS;
-            lockInfo.setTimeoutSeconds((int) timeout);
             performSessionBehavior = true;
         }
-        
+
+        validateLockTimeout(timeout);
+        lockInner(nodeRef, lockInfo, timeout);
+
+        if (performSessionBehavior)
+        {
+            performLockSessionBehavior(nodeRef);
+        }
+    }
+
+    private void validateLockTimeout(int timeout) {
+        if (timeout != WebDAV.TIMEOUT_INFINITY && timeout == LockService.TIMEOUT_INFINITY) {
+            throw new IllegalArgumentException("Timeout == " + LockService.TIMEOUT_INFINITY +
+                    " is treated as permanence for locks. If this lock were intended to be permanent, " +
+                    "it should have the value of a proper static variable");
+        }

```

Comment:
I've reworded the message to be more accurate. "For maximum allowed timeout" since webDav and vti seem to have timeout limited to 24 hours anyway (no option for permanence).